clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/AbstractKafkaConfigProperties.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.kafka;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Common configuration properties for Kafka clients.
+ */
+public abstract class AbstractKafkaConfigProperties {
+
+    protected final Logger log = LoggerFactory.getLogger(getClass());
+
+    /**
+     * Common client config properties.
+     */
+    protected Map<String, String> commonClientConfig;
+    /**
+     * Client id to be applied if no {@code client.id} property is set in the common
+     * or specific client config properties.
+     */
+    protected String clientIdFallback;
+
+    /**
+     * Sets the common Kafka client config properties to be used.
+     *
+     * @param commonClientConfig The config properties.
+     * @throws NullPointerException if the config is {@code null}.
+     */
+    public final void setCommonClientConfig(final Map<String, String> commonClientConfig) {
```

Comment:
Having this abstract base class for all types of Kafka client configuration makes a lot of sense to me.

I'm not convinced, however, that the introduction of a common section where you configure properties for _all_ producers and consumers in an application provides many benefits, but it definitely makes code and documentation more complex and you need to have more knowledge for the configuration. Then there are only a few properties that can be set on both and having them separately, even if they could be equal, does not hurt.

Reply:
My intention was mainly to prevent having to have three identical sets of `bootstrap.servers` and `ssl.*` property sections in each protocol adapter properties. It's about one and the same Kafka cluster here anyhow (and authentication would also be the same, at least for producer and consumer, I guess).
Usage of the common properties is also optional, so, for a configuration with only one of the client types, it doesn't have to be used.